assemblyline/odm/models/ontology/filetypes/macho.py

```diff
@@ -0,0 +1,77 @@
+        is_position_independent = odm.Optional(odm.Boolean())
+        is_norized = odm.Optional(odm.Boolean())
+        uuid = odm.Optional(odm.EmptyableKeyword(copyto="__text__"))
+
+    @odm.model(index=True, store=False)
+    class Segments(odm.Model):
+        offset = odm.Optional(odm.Integer())
+        size = odm.Optional(odm.Integer())
+        flags = odm.Optional(odm.Integer())
+        index = odm.Optional(odm.Integer())
+        numberof_sections = odm.Optional(odm.Integer())
+        name = odm.Optional(odm.EmptyableKeyword(copyto="__text__"))
+        virtual_address = odm.Optional(odm.Integer())
+        virtual_size = odm.Optional(odm.Integer())
+        initial_protection = odm.Optional(odm.EmptyableKeyword(copyto="__text__"))
+        max_protection = odm.Optional(odm.EmptyableKeyword(copyto="__text__"))
+
+    @odm.model(index=True, store=False)
+    class Sections(odm.Model):
+        type = odm.Optional(odm.EmptyableKeyword(copyto="__text__"))
+        name = odm.Optional(odm.EmptyableKeyword(copyto="__text__"))
+        flags = odm.Optional(odm.EmptyableKeyword(copyto="__text__"))
+        alignment = odm.Optional(odm.Integer())
+        relocation_offset = odm.Optional(odm.Integer())
+        numberof_relocations = odm.Optional(odm.Integer())
+        entropy = odm.Optional(odm.Float())
+        segment_name = odm.Optional(odm.EmptyableKeyword(copyto="__text__"))
+        reserved1 = odm.Optional(odm.Integer())
+        reserved2 = odm.Optional(odm.Integer())
+        reserved3 = odm.Optional(odm.Integer())
+        hash = odm.Optional(odm.EmptyableKeyword(copyto="__text__"))
+
+    @odm.model(index=True, store=False)
+    class Libraries(odm.Model):
+        name = odm.Optional(odm.EmptyableKeyword(copyto="__text__"))
+        timestamp = odm.Optional(odm.Integer())
+        compatibility_version = odm.Optional(odm.EmptyableKeyword(copyto="__text__"))
+        current_version = odm.Optional(odm.EmptyableKeyword(copyto="__text__"))
+
+    @odm.model(index=True, store=False)
+    class LoadCommands(odm.Model):
+        name = odm.Optional(odm.EmptyableKeyword(copyto="__text__"))
+        offset = odm.Optional(odm.Integer())
+        size = odm.Optional(odm.Integer())
+
+    @odm.model(index=True, store=False)
+    class Notarization(odm.Model):
+        flags = odm.Optional(odm.EmptyableKeyword(copyto="__text__"))
+        hash_offset = odm.Optional(odm.EmptyableKeyword(copyto="__text__"))
+        hash_size = odm.Optional(odm.EmptyableKeyword(copyto="__text__"))
+        platform = odm.Optional(odm.EmptyableKeyword(copyto="__text__"))
+        codedir_size = odm.Optional(odm.Integer())
+        identifier_string = odm.Optional(odm.EmptyableKeyword(copyto="__text__"))
+        version = odm.Optional(odm.EmptyableKeyword(copyto="__text__"))
+        entitlement = odm.Optional(odm.EmptyableKeyword(copyto="__text__"))
+        requirement = odm.Optional(odm.EmptyableKeyword(copyto="__text__"))
```

Comment:
I believe the root MachO model needs to refer the inner models to be able to use them.
If you can validate that most should be lists except header and notarization, something like the following could solve the issue:
```suggestion
        requirement = odm.Optional(odm.EmptyableKeyword(copyto="__text__"))
        
    header = odm.Optional(odm.Compound(Header))
    segments = odm.Optional(odm.List(odm.Compound(Segments)))
    sections = odm.Optional(odm.List(odm.Compound(Sections)))
    libraries = odm.Optional(odm.List(odm.Compound(Libraries)))
    loadcommands = odm.Optional(odm.List(odm.Compound(LoadCommands)))
    notarization = odm.Optional(odm.Compound(Notarization))
```